tensorflow/python/keras/layers/recurrent_v2.py

```diff
@@ -1333,6 +1336,36 @@ def cudnn_lstm(inputs, init_h, init_c, kernel, recurrent_kernel, bias, mask,
   # so that mathematically it is same as the canonical LSTM implementation.
   full_bias = array_ops.concat((array_ops.zeros_like(bias), bias), 0)
 
+  if build_info.is_rocm_build:
+    units = int(int(kernel.get_shape()[1]) / 4)
+    # ROCm MIOpen's weight sequence for LSTM is different from both canonical
+    # and Cudnn format
```

Comment:
Seems that Rocm is a different implementation then all existing setting, then reusing the cudnn_lstm  seems to be a bad practice. Also, how is the Rocm using cudnn_rnn kernel to work?